### ISSUE TITLE
add_flags causes AttributeError: 'SimulationNormalisation' object has no attribute 'beta_ref'

### ISSUE THREAD

**Issue by theo-brown** [bpatel2107/pyrokinetics#208]
There's a bug somewhere in `pyro.add_flags` that causes an AttributeError.

Running `examples/example_addflags.py` raises the following:
```python
Traceback (most recent call last):
  File "example_addflags.py", line 37, in <module>
    pyro.add_flags(flags)
  File ".../pyrokinetics/pyrokinetics/pyro.py", line 619, in add_flags
    self._check_beta_consistency()
  File ".../pyrokinetics/pyrokinetics/pyro.py", line 1403, in _check_beta_consistency
    f"Explicitly set value of beta ({beta.to(self.norms)}) is inconsistent with "
  File .../pyrokinetics/pyrokinetics/units.py", line 104, in to
    value = as_physical.to(self._convert_base_units(other))
  File ".../pyrokinetics/pyrokinetics/units.py", line 90, in _convert_base_units
    unit = str(getattr(norm, new_unit))
AttributeError: 'SimulationNormalisation' object has no attribute 'beta_ref'
```

**Comment by theo-brown** [comment]
Short-term workaround: set the flags using the `gk_input.data` dict directly.

```python
pyro.gk_input.data["gs2_diagnostics_knobs"] = {
        "write_fields": True,
        "write_kpar": True,
    }
```

**Comment by bpatel2107** [comment]
This will be fixed in #113 (similar issue with normalisation fixed there)
